### PR TITLE
add numba example to enhancingperf.rst

### PR DESCRIPTION
**Pull request by jreback**
enhancing performance pandas 0 16 1-81-g676cb95 documentation 1

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
+1 !

With the latest numba, I think you can do the empty array allocation inside the jitted function, instead of passing it as an argument. I find this a bit easier to follow, but of course in this way it also works with older versions.

File: doc/source/enhancingperf.rst [hunk 1]
```diff
@@ -286,12 +286,66 @@ Even faster, with the caveat that a bug in our cython code (an off-by-one error,
...
+
+    In [4]: %timeit compute_numba(df)
+    1000 loops, best of 3: 774 us per loop
+
+Compared to the ``cython`` example, note that we need to allocate (and pass in) a correctly sized numpy array for the results.
+This will be possible to be done inside the numba-ized function in future versions.
 
-- Loading C modules into cython.
 
-Read more in the `cython docs <http://docs.cython.org/>`__.
```
Comment (jorisvandenbossche) [inline doc/source/enhancingperf.rst]: Can you leave this link at the end of the cython section?

**Comment by jreback** [comment]
@jorisvandenbossche for some reason, sphinx wouldn't render the decorators in the ipython blocks (e.g. the ``@jit``. very odd. (it would work with verbatim, but then you have to repeat the entire code in a suppress block to actually have it executed).

File: doc/source/enhancingperf.rst (hunk 1 above)
Comment (jreback) [inline doc/source/enhancingperf.rst]: links already at the top of the section

File: doc/source/enhancingperf.rst [hunk 2]
```diff
@@ -286,12 +286,66 @@ Even faster, with the caveat that a bug in our cython code (an off-by-one error,
...
+      n = len(col_N)
+      assert len(col_a) == len(col_b) == n
+      for i in range(n):
+         result[i] = integrate_f_numba(col_a[i], col_b[i], col_N[i])
+      return result
+
+    def compute_numba(df):
+       result = np.empty(len(df), dtype='float64')
+       apply_integrate_f_numba(df['a'].values,df['b'].values,df['N'].values,result)
+       return Series(result,index=df.index,name='result')
```
Comment (jorisvandenbossche) [inline doc/source/enhancingperf.rst]: small remark: PEP8, spaces after comma's

**Comment by jreback** [comment]
@jorisvandenbossche I put the allocation back. I think its ok. In the next version of numba, this will just be 'faster' (e.g. its all done in nopython mode). But seems to work ok.

File: doc/source/enhancingperf.rst [hunk 3]
```diff
@@ -286,12 +286,62 @@ Even faster, with the caveat that a bug in our cython code (an off-by-one error,
...
+    def integrate_f_numba(a, b, N):
+       s = 0
+       dx = (b - a) / N
+       for i in range(N):
+           s += f_plain(a + i * dx)
+       return s * dx
+
+    @numba.jit
+    def apply_integrate_f_numba(col_a, col_b, col_N):
+      result = np.empty(len(df), dtype='float64')
```
Comment (jorisvandenbossche) [inline doc/source/enhancingperf.rst]: `df` is not defined here, can use `n` probably

**Comment by seibert** [comment]
Note that the version of Numba that will be able to do the memory allocation in nopython mode will be coming out today.

**Comment by jreback** [comment]
@seibert hmm, this is seeming to execute in nopython mode already though? (I guess except for the allocation, which prob doesn't matter in perf as its a single event), right?

**Comment by jorisvandenbossche** [comment]
It does not execute in nopython mode, but because of the loop lifting of the main part of the function, it does not make that much of a difference.

@seibert yes, I was referring to that version, but forgot that it was only a beta release announcement that I saw

**Comment by seibert** [comment]
Right, loop-lifting means that allocating memory in nopython mode is not as exciting as you might initially think.  :)

**Comment by shoyer** [comment]
@jreback Could you replace the cython timings with a `:verbatim:`, too? Otherwise the numbers will get out of sync.

Generally this looks good to me

**Comment by jreback** [comment]
@shoyer updated to use all verbatims..

**Pull request merged**